### DISCUSSION TITLE
Oqtane with Cosmos EFCore 5 SQL API

### DISCUSSION THREAD

**Discussion by jiggyswift** [oqtane/oqtane.framework#909] [Q&A]
I was just wondering if anyone has tried using Oqtane 2 with Azure Cosmos DB (using the EFCore5 SQL API)?  

I have not even tried yet, but just wanted to reach out to see if this is something I could expect to trivially work?  Or do you guys know of any issues I might hit?  

I have used Cosmos directly myself using the DocumentDB (now SQL API), but know EFcore5 supports it now, so thought it worth asking if this might work.

Thanks,
David

**Comment by hishamco** [comment]
This is similar to https://github.com/oqtane/oqtane.framework/issues/24

**Reply by sbwalker** [reply]
This is essentially a duplicate of #883 - running Oqtane on an alternate data store.

**Comment by amdav** [comment]
I note a few months ago Cosmos launched a "pay as you go" serverless option, making it more viable for products like Oqtane.

**Comment by markdav-is** [comment]
We've used Cosmos as an alternate datasource for the app data.  We stick with the supported DBs for the Oqtane data proper, but for app data, we use Cosmos.  The SQL and EF support in Cosmos is not as plug-and-play from other SQL variants.  Oqtane data is by nature relational, But your app might have the need for non-sql data.
